lib/matplotlib/axes/_axes.py

```diff
@@ -3985,6 +3986,13 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
 
         if c_array is None:
             colors = c     # must be acceptable as PathCollection facecolors
+            try:
+                mcolors.to_rgba_array(colors)
+            except ValueError:
+                # c not acceptable as PathCollection facecolor
+                msg = ("c of shape {0} not acceptable as a color sequence "
+                       "for x with shape {1}, y with shape {2}")
+                raise ValueError(msg.format(c.shape, x.shape, y.shape))
```

Comment:
I see possibilities for improvement in this block: first, the error message is based on the raveled x and y shapes, not the original shapes, so this could be confusing.   Maybe just say 'for x with size...' etc. and use `x.size` etc. After all, x and y are documented as 1-D and treated as 1-D, so it is their size that matters, not their shapes. Second, I think that you could make it both more compact and more efficient by keeping the work done by `mcolors.to_rgba_array`.  You can delete the `colors = c`, and replace the line inside the 'try' with 'colors = mcolors.to_rgba_array(c)'.  It will be subject to another `to_rgba_array` call inside PathCollection, which will be able to recognize that it already has an rgba array, instead of using a list comprehension with a function call for each color.